proxy/src/main.rs

```diff
@@ -26,9 +26,39 @@ mod server_warp;
 /// Origin of data required like the on-chain Registry.
 mod source;
 
+/// Convenience function for setting env vars only if they are unset
+fn set_if_unset (key: &str, val: &str) {
+    match std::env::var(key) {
```

Comment:
You can shorten this with the help of `unwrap_or_else` which executs the closure given in the case the result is an `Err`. More details [here](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else).

Reply:
Great!